ccips/ccip-009/ccip-009-citycoins-vrf-v2.md

```diff
@@ -0,0 +1,214 @@
+
+```
++----------------------+----------+------------+------------+-----------+
+|                      | Consumed | Limit      | Percent    | Diff (v1) |
++----------------------+----------+------------+------------+-----------+
+| Runtime              | 4782     | 5000000000 | 0.000%     | -0.008%   |
++----------------------+----------+------------+------------+-----------+
+| Read count           | 4        | 7750       | 0.052%     | same      |
++----------------------+----------+------------+------------+-----------+
+| Read length (bytes)  | 2220     | 100000000  | 0.002%     | -0.001%   |
++----------------------+----------+------------+------------+-----------+
+| Write count          | 0        | 7750       | 0.000%     | same      |
++----------------------+----------+------------+------------+-----------+
+| Write length (bytes) | 0        | 15000000   | 0.000%     | same      |
++----------------------+----------+------------+------------+-----------+
+```
+
+#### Claim Mining Reward
+
+v1 `claim-mining-reward`
+
+```
++----------------------+----------+------------+------------+
+|                      | Consumed | Limit      | Percent    |
++----------------------+----------+------------+------------+
+| Runtime              | 527214   | 5000000000 | 0.011%     |
++----------------------+----------+------------+------------+
+| Read count           | 32       | 7750       | 0.413%     |
++----------------------+----------+------------+------------+
+| Read length (bytes)  | 58176    | 100000000  | 0.058%     |
++----------------------+----------+------------+------------+
+| Write count          | 5        | 7750       | 0.065%     |
++----------------------+----------+------------+------------+
+| Write length (bytes) | 459      | 15000000   | 0.003%     |
++----------------------+----------+------------+------------+
+```
+
+v2 `claim-mining-reward` on first call before value is saved
+
+```
++----------------------+----------+------------+------------+-----------+
+|                      | Consumed | Limit      | Percent    | Diff (v1) |
++----------------------+----------+------------+------------+-----------+
+| Runtime              | 196647   | 5000000000 | 0.004%     | -0.007%   |
++----------------------+----------+------------+------------+-----------+
+| Read count           | 34       | 7750       | 0.439%     | +0.026%   |
++----------------------+----------+------------+------------+-----------+
+| Read length (bytes)  | 57016    | 100000000  | 0.057%     | -0.001%   |
++----------------------+----------+------------+------------+-----------+
+| Write count          | 6        | 7750       | 0.077%     | +0.012    |
++----------------------+----------+------------+------------+-----------+
+| Write length (bytes) | 492      | 15000000   | 0.003%     | same      |
++----------------------+----------+------------+------------+-----------+
+```
+
+v2 `claim-mining-reward` on subsequent calls after value is saved
```

Comment:
```suggestion
v2 `claim-mining-reward` when `rnd` value has been generated and saved by someone before
```
Only one call per block can be successful, therefore we should avoid "subsequent calls" term.

Reply:
Also fixed in https://github.com/citycoins/governance/pull/5/commits/f4f44b8e6e3cc5cb7ef68d215c29c2cf1676f06f